lib/rubocop/rspec/example.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module RSpec
+    # Wrapper for RSpec examples
+    class Example
+      extend RuboCop::NodePattern::Macros
+
+      def_node_matcher :extract_doc_string,     '(send _ _ $str ...)'
+      def_node_matcher :extract_metadata,       '(send _ _ _ $...)'
+      def_node_matcher :extract_implementation, '(block send args $_)'
+
+      def initialize(node)
+        @node = node
+      end
+
+      def doc_string
+        extract_doc_string(definition)
+      end
+
+      def metadata
+        extract_metadata(definition)
+      end
+
+      def implementation
+        extract_implementation(node)
+      end
+
+      def ==(other)
+        node.eql?(other.node)
+      end
+
+      alias eql? ==
```

Comment:
I think aliasing the other way around, and using `eql?` inside `#eql?` will make the code a tiny bit clearer. What do you think?
```rb
def eql(other)
  node.eql?(other.node)
end

alias == eql?
```